packages/app/src/contacts/actions/select-items.action.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { createAsyncThunk, createAction } from "@reduxjs/toolkit"
+import { ContactsEvent } from "App/contacts/constants"
+import { ReduxRootState, RootState } from "App/__deprecated__/renderer/store"
+
+export const selectAllItems = createAsyncThunk<string[]>(
+  ContactsEvent.SelectAllItems,
+  async (_, { getState }) => {
+    const state = getState() as RootState & ReduxRootState
+    const contactIds = state.contacts.collection
+
+    return contactIds
+  }
+)
+
+export const toggleItem = createAsyncThunk<string[], string>(
+  ContactsEvent.ToggleItem,
+  async (payload, { getState }) => {
+    const state = getState() as RootState & ReduxRootState
+    const selectedItems = state.contacts.selectedItems.rows
+
+    if (selectedItems.includes(payload)) {
+      return selectedItems.filter((item) => item !== payload)
+    } else {
+      return [...selectedItems, payload]
+    }
+  }
+)
+
+export const resetAllItems = createAction(ContactsEvent.ResetAllItems)
```

Comment:
Not in this case. This concept came from those two tasks [CP-1494](https://appnroll.atlassian.net/browse/CP-1494) and [CP-1495](https://appnroll.atlassian.net/browse/CP-1495), when it will be implemented this file will be moved to `core` module and transformed to generic solution